### DISCUSSION TITLE
[PUI] vs old UI, missing features

### DISCUSSION THREAD

**Discussion by turist79** [inventree/InvenTree#8578] [Q&A] [answered]
1) In old UI there was an option to see not only the parts for selected subcategory, but also related stock items

![image](https://github.com/user-attachments/assets/6422630e-3924-46c8-a909-e8639c0c8053)

but i dont see such an option in PUI

![image](https://github.com/user-attachments/assets/26be6db8-3f5c-436e-bc7a-b5bba52f99a6)

What is the right way to see all the stock items for the category?

2) there was an option to select multiple parts with checkboxes and perform part actions (changing category, ordering) and printing actions (print lables). There are printing actions and similar available in PUI for locations and stock items but not for parts.

Is this just "work in progress" on the PUI and those features will be reintroduced later (i`m on 0.16.8 currently) or are those changes permanent?

Thanks

**Comment by turist79** [comment]
Another one:
in stock items overview Location was presented by full location path and in PUI it is only the actual location without path, this seriously impacts the usability for those using more than one location with multiple stocks and similar rack/shelf/bin numbering.

Solution: adding both full path and location as two fields available for stock items table view column selection

**Reply by SchrodingersGat** [reply]
> in stock items overview Location was presented by full location path and in PUI it is only the actual location without path, 

Can you please show some comparative screenshots here of what you mean?

**Reply by turist79** [reply]
> > in stock items overview Location was presented by full location path and in PUI it is only the actual location without path,
> 
> Can you please show some comparative screenshots here of what you mean?

![image](https://github.com/user-attachments/assets/45eb2c1b-66fa-49d1-8756-dd8087e7aee0)

vs

![image](https://github.com/user-attachments/assets/82e3dccb-5d1c-4298-b070-5f528d9ad5fc)

PS: should probably check against demo site as well, will do before posting next time.
Is it ok to switch from stable to master via pulling new docker image and running invoke update?

**Comment by SchrodingersGat** [comment]
@turist79 thanks for reporting - I have a fix incoming here - https://github.com/inventree/InvenTree/pull/8579

Please keep reporting any features which are currently missing in PUI - this will help us to ensure that we are "feature complete" for the next release

**Reply by turist79** [reply]
> Please keep reporting any features which are currently missing in PUI

Stock items list is missing Supplier Part column
![image](https://github.com/user-attachments/assets/cb65862b-8d38-47f9-a310-dfa18151872e)

while it was present in old UI (sorry, had to cover some data :) )

![image](https://github.com/user-attachments/assets/543488a5-a691-46d2-b96e-bd96bb922d9a)

**Reply by SchrodingersGat** [reply, marked as answer]
It is available in "master" branch - if you are currently running "stable" you will need to check against the demo site :)

![image](https://github.com/user-attachments/assets/f8759b06-b16c-4e7a-bfc4-e642a1bbd270)

**Reply by turist79** [reply]
> It is available in "master" branch - if you are currently running "stable" you will need to check against the demo site :)
> 

Ohhh i felt like i`m missing something and there should be newer version (actually, i have seen some plugins requiring minimum 0.17 app version which i don`t seem to find anywhere). Version > 0.16.8 is it currently not available as docker container, isn`t it?

Sorry :)

**Reply by wolflu05** [reply]
The :latest docker tag currently contains 0.17 until the release where 0.17 will be available in :stable. Currently 0.16 is stable.

**Reply by turist79** [reply]
Former admin UI allowed both data upload and download for all data models.

![image](https://github.com/user-attachments/assets/ca8a1f26-5057-4ae5-afc7-8b5dc9b3c643)

In PUI i see the option to upload data only

![image](https://github.com/user-attachments/assets/e02ed2e5-f7a6-4a9b-9cb3-097f4743d743)

**Reply by wolflu05** [reply]
You can now download the data in the individual PUI tables. E.g. for the stock items:
<img width="1181" alt="image" src="https://github.com/user-attachments/assets/c8e434ad-7252-48d0-b9df-933009cd5cf1">

**Reply by SchrodingersGat** [reply]
Also the admin UI is still present.

**Reply by turist79** [reply]
> Also the admin UI is still present.

How do you access it - is it not the Admin center settings section?
I meant the page under xxx/admin/ which was accessible earliver by the menu

![image](https://github.com/user-attachments/assets/afbb2d55-b0bc-4993-9fa8-b222d6eb2c59)

or by  the special blue "open in admin" button in the actual object (part etc)

![image](https://github.com/user-attachments/assets/576c2281-31ad-4fba-8128-3278791dc41b)

and

![image](https://github.com/user-attachments/assets/ead45cb2-51a7-4877-b323-89019a0c4267)

When i click this button in PUI both in 0.16.8 and demo it opens https://xxx/index/ page eventually even going back to the classic UI.

**Reply by matmair** [reply]
Although uploading data to the Django admin should be considered dangerous as certain business logic constraints are not enforced through it

**Reply by turist79** [reply]
> Although uploading data to the Django admin should be considered dangerous as certain business logic constraints are not enforced through it

Yes, understandable.
The upload via Data import in PUI`s Admin center is considered safe in this regard?

PS: Should "Download data"-generated tables be used as templates for Data import function or there is some proper way to get templates for the datamodels?

**Reply by SchrodingersGat** [reply]
Regarding import and export - this is changing considerably and there will be new documentation added in 1.0.0. 

The current process in the "Django admin" will be removed entirely as it is slow and error prone

**Reply by turist79** [reply]
Updated to latest 0.17.0 image, thank for the advice.
There is still missing column in stock items list - Notes.
We use notes it to write down comments related to the item status etc. It was convinient in old UI to have short notes right in the stock items overview.
